File: front-angular/src/app/modelos/material.model.ts
export class Material {
    id?: number;
    nombreMaterial: string | undefined | null;
    numeroPuntos?: number;
   

    constructor(id?: number, nombreMaterial?: string, numeroPuntos?: number
        ) {
        this.id = id;
        this.nombreMaterial = nombreMaterial;
        this.numeroPuntos = numeroPuntos;
    }
}

File: reciward_flutter_app/lib/features/material/data/models/material_model.dart

import 'package:reciward_flutter_app/features/material/domain/entities/material_entity.dart';

class MaterialModel {

  final String ? id;
  final String ? nombreMaterial;
  final int ? numeroPuntos;
  final String ? clasificacion;


  const MaterialModel({this.id, this.clasificacion, this.nombreMaterial, this.numeroPuntos});

  factory MaterialModel.fromJson(Map<String, dynamic> json){
    return MaterialModel(
      id: json['id'].toString(),
      nombreMaterial: json['nombreMaterial'],
      numeroPuntos: json['numeroPuntos'],
      clasificacion: json['clasificacion']
    );
  }

  MaterialEntity toEntity(){
    return MaterialEntity(
      clasificacion: clasificacion,
      id: id,
      nombreMaterial: nombreMaterial,
      numeroPuntos: numeroPuntos
    );
  }
}